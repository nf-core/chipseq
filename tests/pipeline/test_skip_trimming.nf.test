nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    tag "test_skip_trimming"
    tag "pipeline"

    test("Skip Trimming") {

        when {
            params {
                outdir        = "$outputDir"
                skip_trimming = true
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir")).match("software_versions") },
                { assert snapshot(path("$outputDir/bwa/mergedLibrary/SPT5_INPUT_REP1.mLb.clN.sorted.bam"),
                                path("$outputDir/bwa/mergedLibrary/SPT5_INPUT_REP2.mLb.clN.sorted.bam"),
                                path("$outputDir/bwa/mergedLibrary/SPT5_T0_REP1.mLb.clN.sorted.bam"),
                                path("$outputDir/bwa/mergedLibrary/SPT5_T0_REP2.mLb.clN.sorted.bam"),
                                path("$outputDir/bwa/mergedLibrary/SPT5_T15_REP1.mLb.clN.sorted.bam"),
                                path("$outputDir/bwa/mergedLibrary/SPT5_T15_REP2.mLb.clN.sorted.bam")).match("bwa/mergedLibrary/") },
                { assert new File("$outputDir/bwa/mergedLibrary/bigwig/").exists() },
                { assert new File("$outputDir/bwa/mergedLibrary/deepTools/").exists() },
                { assert new File("$outputDir/bwa/mergedLibrary/macs2/").exists() },
                { assert new File("$outputDir/bwa/mergedLibrary/phantompeakqualtools/").exists() },
                { assert new File("$outputDir/bwa/mergedLibrary/picard_metrics/").exists() },
                { assert new File("$outputDir/bwa/mergedLibrary/samtools_stats/").exists() },
                { assert new File("$outputDir/fastqc/").exists() },
                { assert snapshot(path("$outputDir/genome").list()).match("genome") },
                { assert new File("$outputDir/igv/").exists() },
                { assert new File("$outputDir/multiqc/").exists() },
                { assert new File("$outputDir/pipeline_info/").exists() },
                { assert !new File("$outputDir/trimgalore/").exists() }
            )
        }

    }

}
