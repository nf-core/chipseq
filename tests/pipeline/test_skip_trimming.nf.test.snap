{
    "genome": {
        "content": [
            "genome.fa:md5,4bad9f4b18056156a81f7f952abbe125",
            "genome.fa.fai:md5,6f4c0ce5258e6948135ad006e1f9ee1b",
            "genome.fa.sizes:md5,e122f8021a5ef14104c32785cb9b7adc",
            "genome.include_regions.bed:md5,58b27f8faf81d116a596c37be421ce01"
        ],
        "timestamp": "2023-06-25T23:55:28+0000"
    },
    "software_versions": {
        "content": [
            "{ANNOTATE_BOOLEAN_PEAKS={sed=4.7}, BAMTOOLS_FILTER={bamtools=2.5.2, samtools=1.15.1}, BAM_REMOVE_ORPHANS={samtools=1.15.1}, BEDTOOLS_GENOMECOV={bedtools=2.30.0}, BWA_INDEX={bwa=0.7.17-r1188}, BWA_MEM={bwa=0.7.17-r1188, samtools=1.16.1}, CUSTOM_DUMPSOFTWAREVERSIONS={python=3.11.0, yaml=6.0}, CUSTOM_GETCHROMSIZES={getchromsizes=1.16.1}, DEEPTOOLS_COMPUTEMATRIX={deeptools=3.5.1}, DEEPTOOLS_PLOTFINGERPRINT={deeptools=3.5.1}, DEEPTOOLS_PLOTHEATMAP={deeptools=3.5.1}, DEEPTOOLS_PLOTPROFILE={deeptools=3.5.1}, FASTQC={fastqc=0.11.9}, FRIP_SCORE={bedtools=2.30.0, samtools=1.15.1}, GENOME_BLACKLIST_REGIONS={bedtools=2.30.0}, GTF2BED={perl=5.26.2}, HOMER_ANNOTATEPEAKS_CONSENSUS={homer=4.11}, HOMER_ANNOTATEPEAKS_MACS2={homer=4.11}, IGV={python=3.8.3}, MACS2_CALLPEAK={macs2=2.2.7.1}, MACS2_CONSENSUS={python=3.10.0, r-base=4.1.1}, PHANTOMPEAKQUALTOOLS={phantompeakqualtools=1.2.2}, PICARD_COLLECTMULTIPLEMETRICS={picard=3.0.0}, PICARD_MARKDUPLICATES={picard=3.0.0}, PICARD_MERGESAMFILES={picard=3.0.0}, PLOT_HOMER_ANNOTATEPEAKS={r-base=4.0.3}, PLOT_MACS2_QC={r-base=4.0.3}, SAMPLESHEET_CHECK={python=3.8.3}, SAMTOOLS_FLAGSTAT={samtools=1.17}, SAMTOOLS_IDXSTATS={samtools=1.17}, SAMTOOLS_INDEX={samtools=1.17}, SAMTOOLS_SORT={samtools=1.17}, SAMTOOLS_STATS={samtools=1.17}, SUBREAD_FEATURECOUNTS={subread=2.0.1}, UCSC_BEDGRAPHTOBIGWIG={ucsc=377}, Workflow={nf-core/chipseq=2.1.0dev}}"
        ],
        "timestamp": "2023-06-25T23:55:28+0000"
    },
    "bwa/mergedLibrary/": {
        "content": [
            "SPT5_INPUT_REP1.mLb.clN.sorted.bam:md5,78d3f645ed745f164ca42cce6942086c",
            "SPT5_INPUT_REP2.mLb.clN.sorted.bam:md5,f584d049da692581f0cafa75c6adf469",
            "SPT5_T0_REP1.mLb.clN.sorted.bam:md5,927de7ba2cb30ee8235f273b3c4a86fc",
            "SPT5_T0_REP2.mLb.clN.sorted.bam:md5,9e80eaea382f9fbfd8ce000d1130785f",
            "SPT5_T15_REP1.mLb.clN.sorted.bam:md5,0899b798da9eb097a5b3fe29f3e5a056",
            "SPT5_T15_REP2.mLb.clN.sorted.bam:md5,25707d4184d1a4a9c11bc52f65f1ce03"
        ],
        "timestamp": "2023-06-25T23:55:28+0000"
    }
}